File: 2487.py
class Solution:
    def partitionString(self, s: str) -> int:
        n = len(s)
        i = 0
        res = 0
        while i < n:
            temp = {}
            for j in range(i, n):
                idx = s[j]
                if idx not in temp:
                    temp[idx] = 1
                    i += 1
                else:
                    break
            res += 1
        
        return res

File: 2487.cpp
class Solution {
public:
    int partitionString(string s) {
        int n = s.size();
        int res = 0;
        set<char> c;
        for (char i : s){
            if (c.find(i) != c.end()){
                res += 1;
                c.clear();
            }
            c.insert(i);
        }
        return res + 1;
    }
};